src/critics/smoothness_critic.cpp

```diff
@@ -0,0 +1,54 @@
+// Copyright 2022 @artofnothingness Alexey Budyakov, Samsung Research
+#include "mppic/critics/smoothness_critic.hpp"
+
+namespace mppi::critics
+{
+
+void SmoothnessCritic::initialize()
+{
+  auto getParam = parameters_handler_->getParamGetter(name_);
+
+  getParam(vx_power_, "vx_cost_power", 1);
+  getParam(vx_weight_, "vx_cost_weight", 0.1);
+
+  getParam(vy_power_, "vy_cost_power", 1);
+  getParam(vy_weight_, "vy_cost_weight", 0.1);
+
+  getParam(wz_power_, "wz_cost_power", 1);
+  getParam(wz_weight_, "wz_cost_weight", 0.1);
+}
+
+void SmoothnessCritic::score(models::CriticFunctionData & data)
+{
+  using namespace xt::placeholders;  // NOLINT
+  
+  if (!enabled_) {
+    return;
+  }
+
+  auto vx = data.state.getControlVelocitiesVX();
+  auto wz = data.state.getControlVelocitiesWZ();
+
+  auto vx_diff = xt::abs(xt::view(vx, xt::all(), xt::range(1, _)) - xt::view(vx, xt::all(), xt::range(0, -1)));
```

Comment:
Forgive my ignorance, but on L33 you use `wz` and then on L35/36/42/43 I see a ix of `vx` and `wz`'s when I think they should be `vy` and such. It seems like maybe a copy paste error?

Reply:
fixed